.github/workflows/integration.yml

```
@@ -36,22 +36,30 @@ jobs:
       - name: Install nightly Rust
         run: rustup toolchain install nightly
   
-      - name: Install rustfmt for nightly
-        run: rustup component add rustfmt --toolchain nightly
+      - name: Install testing tools
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-llvm-cov
+
+      - name: Install lcov
+        run: sudo apt-get update && sudo apt-get install lcov
 
       - name: Cache Rust dependencies
         uses: Swatinem/rust-cache@v2
         with:
           cache-on-failure: true
-      - name: clippy
-        uses: dtolnay/rust-toolchain@clippy
-        
-      - name: Install Foundry
-        run: cargo install --git https://github.com/foundry-rs/foundry --profile release --locked forge cast chisel anvil
 
       - name: docker
         uses: docker-practice/actions-setup-docker@master
 
-      - name: Run make pr
-        run: make pr
+      - name: Run tests and generate code coverage
+        run: make coverage
 
+      - name: Generate coverage summary
+        run: lcov --summary lcov.info | tee coverage-summary.txt
+
+      - name: Upload coverage report
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-summary
+          path: ./coverage-summary.txt
```

I don't think this action is worth it. We should either print the summary on the workflow logs or upload the complete coverage report as an artifact.

---

An alternative would be to use something like CodeCov:
```suggestion
      - name: Upload coverage report
        uses: codecov/codecov-action@v4
        with:
          files: '*.info'
          fail_ci_if_error: true
```

---

Apparently CodeCov doesn't work without the token so for now i will upload the whole lcov.info with upload-artifact. It will be accesible from the CI logs